src/percy-agent-client/percy-agent.ts

```diff
@@ -6,41 +6,53 @@ export default class PercyAgent {
   clientInfo: string | null
   environmentInfo: string | null
   xhr: any
+  postSnapshotDirectly: boolean
   port: number
   domTransformation: any | null
-  client: PercyAgentClient
+  client: PercyAgentClient | null
 
   readonly defaultDoctype = '<!DOCTYPE html>'
 
   constructor(options: ClientOptions = {}) {
     this.clientInfo = options.clientInfo || null
     this.environmentInfo = options.environmentInfo || null
-    this.xhr = options.xhr || XMLHttpRequest
+    // Default to 'true' unless explicitly disabled.
+    this.postSnapshotDirectly = options.postSnapshotDirectly !== false
     this.domTransformation = options.domTransformation || null
     this.port = options.port || 5338
 
-    this.client = new PercyAgentClient(
-      `http://localhost:${this.port}`,
-      this.xhr,
-    )
+    if (this.postSnapshotDirectly) {
+      this.xhr = options.xhr || XMLHttpRequest
+      this.client = new PercyAgentClient(
+        `http://localhost:${this.port}`,
+        this.xhr,
+      )
+    } else {
+      this.xhr = null
```

Comment:
Do we need to actually set these as null? I would assume that's their default value.

If not that can be done inline with the definition. E.g.

```
client: PercyAgentClient | null = null
```